### PR TITLE
Checked rekor type handling

### PR DESCRIPTION
**Pull request by loosebazooka**
Adjusts old rekor type wrapper to now be aware of the spec kind and api version.

Signed-off-by: Appu Goundan <appu@google.com>

### CODE REVIEW

**Review by patflynn — COMMENTED** [review]
If I understand this correctly we foresee other types being returned for the RekorEntry and we want to be able to handle them?

File: src/main/java/dev/sigstore/rekor/client/RekorEntry.java [hunk 1]
```diff
@@ -69,11 +68,15 @@ interface InclusionProof {
...
   String getBody();
 
+  /**
+   * Returns a decoded {@link RekorEntryBody} of the log entry. Use {@link RekorTypes} to further
+   * process.
+   */
   @Value.Derived
-  default HashedRekord getBodyAsHashedrekord() {
+  default RekorEntryBody getBodyDecoded() {
+    System.out.println(new String(Base64.getDecoder().decode(getBody()), UTF_8));
```
[the remark this answers is not in the document]
Comment (loosebazooka) [reply src/main/java/dev/sigstore/rekor/client/RekorEntry.java:77]: Whoops

[the remark this answers is not in the document]
Comment (loosebazooka) [reply src/main/java/dev/sigstore/rekor/client/RekorEntry.java]: done

**Comment by loosebazooka** [comment]
> If I understand this correctly we foresee other types being returned for the RekorEntry and we want to be able to handle them?

Yeah, this also lets us handle a situation in the library when someone tries to verify something we weren't designed to do.

**Review by patflynn — APPROVED** [review]

**Pull request merged**